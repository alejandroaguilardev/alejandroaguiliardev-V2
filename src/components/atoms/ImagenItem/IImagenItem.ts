import { IDataGalleryEducation } from "../../../data/dataGalleryEducation";

export interface IImagenItem {
	data: IDataGalleryEducation;
	setOpen: (open: boolean) => void;
	setItem: (item: IDataGalleryEducation) => void;
}
