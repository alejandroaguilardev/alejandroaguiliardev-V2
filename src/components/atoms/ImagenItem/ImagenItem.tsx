import Image from "next/image";
import { ImageListItem, ImageListItemBar, IconButton } from "@mui/material"
import InfoIcon from '@mui/icons-material/Menu';
import { IImagenItem } from "./IImagenItem";

export const ImagenItem = ({ data,  setOpen,  setItem }: IImagenItem) => {
    return (
        <ImageListItem
            key={data.img}
            onClick={() => { setOpen(true); setItem({ ...data }) }}
        >
            <Image
                src={data.img}
                alt={data.title}
                width={400}
                height={300}
            />
            <ImageListItemBar
                title={data.title}
                subtitle={data.author}
                actionIcon={
                    <IconButton
                        sx={{ color: 'rgba(255, 255, 255, 0.54)' }}
                        aria-label={`info about ${data.title}`}
                    >
                        <InfoIcon />
                    </IconButton>
                }
            />
        </ImageListItem>
    )
}
