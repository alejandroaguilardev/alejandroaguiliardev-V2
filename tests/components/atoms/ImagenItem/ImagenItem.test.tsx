import { render } from "@testing-library/react";
import { ImagenItem } from "../../../../src/components/atoms/ImagenItem/ImagenItem";

describe("ImagenItem test", () => {
	const data = {
		img: "/img/alejandro-aguilar-dev.svg",
		title: "A title",
		author: "Alejandro",
	};

	test("ImagenItem Snapshot", () => {
		const mockSetOpen = jest.fn();
		const mocksetItem = jest.fn();

		const { container } = render(
			<ImagenItem data={data} setOpen={mockSetOpen} setItem={mocksetItem} />
		);

		expect(container).toMatchSnapshot();
	});

	// test("ImagenItem Props", () => {
	// 	render(
	// 		<ImagenItem data={data} setOpen={mockSetOpen} setItem={mocksetItem} />
	// 	);
	// });
});
