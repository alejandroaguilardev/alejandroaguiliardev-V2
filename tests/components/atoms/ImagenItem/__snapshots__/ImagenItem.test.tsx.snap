// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagenItem test ImagenItem Snapshot 1`] = `
<div>
  <li
    class="MuiImageListItem-root css-l086p4-MuiImageListItem-root"
    style="height: auto; grid-column-end: span 1; grid-row-end: span 1;"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27300%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="A title"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    <div
      class="MuiImageListItemBar-root MuiImageListItemBar-positionBottom css-ztrqiq-MuiImageListItemBar-root"
    >
      <div
        class="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapBottom MuiImageListItemBar-titleWrapActionPosRight css-14q7c0c-MuiImageListItemBar-titleWrap"
      >
        <div
          class="MuiImageListItemBar-title css-dasnyc-MuiImageListItemBar-title"
        >
          A title
        </div>
        <div
          class="MuiImageListItemBar-subtitle css-hk6gwf-MuiImageListItemBar-subtitle"
        >
          Alejandro
        </div>
      </div>
      <div
        class="MuiImageListItemBar-actionIcon MuiImageListItemBar-actionIconActionPosRight css-mjbero-MuiImageListItemBar-actionIcon"
      >
        <button
          aria-label="info about A title"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1lujp9z-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </li>
</div>
`;
